lib/vanagon/component/source/git.rb

```diff
@@ -1,39 +1,69 @@
 require 'vanagon/utilities'
+require 'vanagon/errors'
+# This stupid library requires a capital 'E' in its name
+# but it provides a wealth of useful constants
+require 'English'
+require 'fustigit'
+require 'git/unshallow_repositories'
+require 'git/basic_submodules'
+require 'logger'
 
 class Vanagon
   class Component
     class Source
       class Git
-        include Vanagon::Utilities
-        attr_accessor :url, :ref, :workdir, :version, :cleanup
+        attr_accessor :url, :ref, :workdir, :depth
+        attr_reader :version, :default_options, :repo
+
+        class << self
+          # Attempt to connect to whatever URL is provided and
+          # return True or False depending on whether or not
+          # `git` thinks it's a valid Git repo.
+          #
+          # @return [Boolean] whether #url is a valid Git repo or not
+          def valid_remote?(url)
+            !!::Git.ls_remote(url)
+          rescue ::Git::GitExecuteError
+            nil
```

Comment:
Since you're making it a boolean (in the yard, and explicitly above), should this return `false` instead?